man/coap_attribute.txt.in

```diff
@@ -0,0 +1,141 @@
+// -*- mode:doc; -*-
+// vim: set syntax=asciidoc,tw=0:
+
+coap_attribute(3)
+=================
+:doctype: manpage
+:man source:   coap_attribute
+:man version:  @PACKAGE_VERSION@
+:man manual:   libcoap Manual
+
+NAME
+----
+coap_attribute, coap_add_attr, coap_find_attr
+- work with CoAP attributes
+
+SYNOPSIS
+--------
+*#include <coap/coap.h>*
+
+*coap_attr_t *coap_add_attr(coap_resource_t *_resource_, const unsigned char 
+*_name_, size_t _nlen_, const unsigned char *_value_, size_t _vlen_, 
+int _flags_);*
+
+*coap_attr_t *coap_find_attr(coap_resource_t *_resource_, const unsigned 
+char *_name_, size_t _nlen_);*
+
+Link with -lcoap-1.
+
+DESCRIPTION
+-----------
+CoAP Resources on a CoAP Server need to be created, updated etc. The URI in 
+the request packet defines the resource to work with, with possibly the Query 
+referring to a sub-resource.
+When resources are configured on the CoAP server, the URI to match against 
+is specified. 
+Callback Handlers are then added to the resource to handle the different 
+request methods.
+Adding Attributes allows textual information to be added to the resource 
+which can then be reported back to any client doing a "GET .well-known/core" 
+request. 
+
+Attributes are automatically freed when a Resource is deleted.
+
+The *coap_add_attr*() function
+registers a new attribute called _name_ with a name length of _nlen_ 
+for the _resource_. 
+The value of the attribute is _value_ with a length of _vlen_.
+_flags_ can be one or more of the following, which, if set, defines what is 
+to be internally freed off when the attribute is deleted with 
+*coap_delete_resource*().
+It is imperative that _name_ and _value_ point to data that is valid during 
+the lifetime of the attribute.
+
+*COAP_ATTR_FLAGS_RELEASE_NAME*::
+Free off _name_ when attribute is deleted with *coap_delete_resource*().
+
+*COAP_ATTR_FLAGS_RELEASE_NAME*::
+Free off _value_ when attribute is deleted with *coap_delete_resource*().
+
+The *coap_find_attr*() function returns the attribute with the _name_ with 
+a length of _nlen_, if found, associated with _resource_.
+
+RETURN VALUES
+-------------
+*coap_add_attr*() function returns a pointer to the attribute that was 
+created or NULL if there is a malloc failure. 
+
+*coap_find_attr*() function returns a pointer to the first matching 
+attribute or NULL if the _name_ was not found. 
+
+EXAMPLE
+-------
+*Initialize Resources*
+
+[source, c]
+--
+#include <coap/coap.h>
+
+static void
+init_resources(coap_context_t *ctx)
+{ 
```

Comment:
I would like the examples to follow the coding guidelines of libcoap. Here, this means that the block-opening parenthesis is at the end of the previous line.

Reply:
As it happens, a2x does that anyway, but I will make the change.  However, there have to be some blank lines to make things format, asthe a2x ' + ' fails tor work for me to start a new line.